src/network-layer/default/RelayDefaultNetworkLayer.js

```diff
@@ -60,9 +60,11 @@ class RelayDefaultNetworkLayer {
       } else {
         request.resolve({response: payload.data});
       }
-    }).catch(
-      error => request.reject(error)
-    );
+      return request;
+    }).catch(error => {
+      request.reject(error);
+      return request;
+    });
```

Comment:
Hmm. It's already possible for users to wait for the results of `request` directly:

```javascript
class MutationLoggingNetworkLayer extends DefaultNetworkLayer {
  sendMutation(mutation) {
    // log the response or error (note that `mutation` is a promise)
    mutation.then(
      response => console.log(response),
      error => console.error(error),
    );
    // Send the mutation using the default network implementation
    return super.sendMutation(mutation);
  }
```

Have you tried this approach?